src/flows/pipes/RawFluxEditor/DynamicFunctions/FunctionsList.tsx

```diff
@@ -0,0 +1,139 @@
+  getFluxPackages: () => void
+}
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = ReduxProps & OwnProps & DispatchProps
+
+const FunctionsList: FC<Props> = (props: Props) => {
+  const [search, setSearch] = useState('')
+  const updateSearch = useCallback(
+    text => {
+      setSearch(text)
+    },
+    [search, setSearch]
+  )
+  const [fluxLoadingState, setfluxLoadingState] = useState<RemoteDataState>(
+    RemoteDataState.NotStarted
+  )
+
+  const {fluxFunctions, getFluxPackages} = props
+
+  useEffect(() => {
+    const getFluxFuncs = async () => {
+      try {
+        setfluxLoadingState(RemoteDataState.Loading)
+
+        if (fluxFunctions.length === 0) {
+          await getFluxPackages()
+          setfluxLoadingState(RemoteDataState.Done)
+        }
+        setfluxLoadingState(RemoteDataState.Done)
+      } catch (err) {
+        console.error(err)
+        setfluxLoadingState(RemoteDataState.Error)
+      }
+    }
+    getFluxFuncs()
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps
+
+  const sortedFunctions = useMemo(
+    () =>
+      fluxFunctions.sort((a, b) => {
+        if (a.package.toLowerCase() === b.package.toLowerCase()) {
+          return a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1
+        } else {
+          return a.package.toLowerCase() < b.package.toLowerCase() ? -1 : 1
+        }
+      }),
+    [fluxFunctions]
+  )
+
+  const filteredFunctions = sortedFunctions.filter(fn => {
+    return (
+      !search.length ||
+      fn.name.toLowerCase().includes(search.toLowerCase()) ||
+      fn.package.toLowerCase().includes(search.toLowerCase())
+    )
+  })
+
+  return useMemo(() => {
+    let fnComponent
+
+    if (filteredFunctions.length === 0) {
+      fnComponent = (
+        <EmptyState size={ComponentSize.ExtraSmall}>
+          <EmptyState.Text>No functions match your search</EmptyState.Text>
+        </EmptyState>
+      )
+    } else {
+      fnComponent = filteredFunctions.map((fn, index) => (
+        <Fn
+          onClick={onselect}
+          extractor={fn => `${fn.package}.${fn.name}`}
+          key={index}
+          option={fn}
+          testID={fn.name}
+          ToolTipContent={FunctionTooltipContent}
+        />
+      ))
+    }
+
+    return (
+      <SpinnerContainer
+        loading={fluxLoadingState}
+        spinnerComponent={<TechnoSpinner />}
+      >
+        <div className="flux-toolbar">
+          <div className="flux-toolbar--search">
+            <SearchWidget
+              placeholderText="Filter Functions..."
+              onSearch={updateSearch}
+              searchTerm={search}
+              testID="flux-toolbar-search--input"
+            />
+          </div>
+          <div className="flux-toolbar--list" data-testid="flux-toolbar--list">
+            {fnComponent}
+          </div>
+        </div>
+      </SpinnerContainer>
+    )
+  }, [filteredFunctions, fluxLoadingState, onselect, search, updateSearch])
```

Comment:
This is a strange pattern. It looks like the function is returning JSX that comes from `useMemo`. I don't think that's necessary, because your dependency array has a lot of variables that will change. There's basically no difference with not wrapping it at all in `useMemo`.

I tested it a bit locally, and I cannot see any performance difference. The number of re-renders is the same because of props and dependencies.

Reply:
I would not wrap the `useMemo` here at all.

Reply:
@wiedld - Yes, the FilterList file is actually part of the Flux Dynamic Help Panels Epic's ongoing work. Let's pair sometime today and I will explain further.